presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -163,6 +164,18 @@ public ListeningExecutorService createFileRanemeExecutor(HiveConnectorId hiveCli
                                 hiveClientConfig.getMaxConcurrentFileRenames())));
     }
 
+    @ForFileCreate
+    @Singleton
+    @Provides
+    public ListeningExecutorService createFileCreateExecutor(HiveConnectorId hiveClientId, HiveClientConfig hiveClientConfig)
+    {
+        return listeningDecorator(
+                new ExecutorServiceAdapter(
+                        new BoundedExecutor(
+                                newCachedThreadPool(daemonThreadsNamed("hive-" + hiveClientId + "-%s")),
```

Comment:
How about naming these threads `hive-create-zero-row-file...`, and correspondingly naming the threads in rename to be `hive-rename...`. You could change the rename one in a separate commit before the main commit.